pkg/tfgen/docs.go

```diff
@@ -621,6 +622,35 @@ func cleanupText(text string) string {
 		return parts[1]
 	})
 
+	// Fixup resource and property name references
+	text = codeLikeSingleWord.ReplaceAllStringFunc(text, func(match string) string {
+		parts := codeLikeSingleWord.FindStringSubmatch(match)
+		name := parts[2]
+		info, hasResourceInfo := g.info.Resources[name]
+		if hasResourceInfo {
+			// This is a resource name
+			resname, mod := resourceName(g.info.GetResourcePrefix(), name, info, false)
+			modname := extractModuleName(mod)
+			switch g.language {
+			case golang, python:
+				// Use `ec2.Instance` format
+				return parts[1] + modname + "." + resname + parts[4]
+			default:
+				// User `aws.ec2.Instance` format`
+				return parts[1] + g.info.GetResourcePrefix() + "." + modname + "." + resname + parts[4]
```

Comment:
Using `g.info.GetResourcePrefix()` returns `google` for the GCP provider, but we generally refer to these using the package name, e.g. `gcp`.

For example:

https://github.com/pulumi/pulumi-gcp/compare/lukehoban/namerewriting#diff-5ecf084a4a91a5264583a5959a8a5f04R125

`google.compute.Image` should be `gcp.compute.Image`.

Reply:
Fixed.